src/orquestra/sdk/_base/_driver/_client.py

```diff
@@ -579,15 +586,61 @@ def get_workflow_run_logs(self, wf_run_id: _models.WorkflowRunID) -> bytes:
             ).dict(),
         )
 
+        # Handle errors
         if resp.status_code == codes.NOT_FOUND:
             raise _exceptions.WorkflowRunLogsNotFound(wf_run_id)
         elif resp.status_code == codes.BAD_REQUEST:
             raise _exceptions.InvalidWorkflowRunID(wf_run_id)
 
         _handle_common_errors(resp)
 
-        # TODO: unzip, get logs (ORQSDK-652)
-        return resp.content
+        # Decompress data
+        try:
+            unzipped: bytes = zlib.decompress(resp.content, 16 + zlib.MAX_WBITS)
+        except zlib.error as e:
+            raise _exceptions.WorkflowRunLogsNotReadable(wf_run_id) from e
+
+        untarred = TarFile(fileobj=io.BytesIO(unzipped)).extractfile("step-logs")
+        assert untarred is not None
+        decoded = r"""{}""".format(untarred.read().decode("utf-8"))
+
+        # Parse the decoded data as logs
+        # TODO: index by taskinvocationID rather than workflowrunID [ORQSDK-777]
+        logs = []
+        for section in decoded.split("\n"):
+            if len(section) < 1:
+                continue
+            for log in json.loads(section):
+                try:
+                    # Orquestra logs are jsonable - where we can we parse these and
+                    # extract the useful information
+                    parsed_log_message = json.loads(log[1]["log"])
+                    interpreted_log = WFLog(
+                        timestamp=datetime.fromtimestamp(log[0], timezone.utc),
+                        level=parsed_log_message["level"],
+                        filename=parsed_log_message["filename"],
+                        message=parsed_log_message["message"],
+                        wf_run_id=parsed_log_message["wf_run_id"],
+                        task_inv_id=None,
+                        task_run_id=parsed_log_message["task_run_id"],
+                    )
+
+                except (json.JSONDecodeError, KeyError):
+                    # If the log isn't jsonable (i.e. it's from Ray) extract what info
+                    # we can.
+                    interpreted_log = WFLog(
+                        timestamp=datetime.fromtimestamp(log[0], timezone.utc),
+                        level="",
+                        filename=log[1]["ray_filename"],
+                        message=log[1]["log"],
+                        wf_run_id=log[1]["tag"],
+                        task_inv_id=None,
+                        task_run_id=None,
+                    )
```

Comment:
> please don't use empty strings (level="")

I take it you didn't intend to suggest it as a sentinel value then? https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/79#discussion_r1141950401 WFLog won't allow a Nonetype there so I'll switch it to a "FIXME"

> our interpretation of WFLog.filename is the name of the file where the logger was used. ray_filename is probably the name of the log file produced by Ray, which is a different string.

Yep good point, that needs to go!

log[1]["tag"] depends on the fluentbit configuration. Are you sure this is always going to be WorkflowRunId?

No, not even slightly, but I can't think of a way to determine from the information in the logs what it actually is!

Reply:
ATM this method produces `WFLog`s. Seems like the only field we're using it for (as of today) is the `message`: https://github.com/zapatacomputing/orquestra-workflow-sdk/blob/c8387d76750ab892454b7d953a07ce7bfeb49acb/src/orquestra/sdk/_base/_driver/_ce_runtime.py#L400

Instead of producing an object with sentinel values, only to discard it soon after, WDYT about just returning a list of strings from this method? See also a patch suggestion in the other comment

Reply:
[b0e2a54](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/79/commits/b0e2a54e8cc345eddb9a5a6ebd4ba307a8ebcc99)